src/Core/Content/Product/SalesChannel/Listing/ProductListingFeaturesSubscriber.php

```diff
@@ -371,7 +371,7 @@ function (array $config) {
         $groups = array_column($groups, 'id');
 
         // expanded group count matches option count? All variants are displayed
-        if (count($groups) === count($product->getOptionIds())) {
+        if (is_array($product->getOptionIds()) && count($groups) === count($product->getOptionIds())) {
             return false;
         }
 
```

Comment:
here is a check for the options. maybe if we put that before the count if, the error will already be fixed?

Reply:
@mitelg : yes, when I applied the fix, I haven't got the issue again.

Reply:
ok, would be nice, if could change your PR accordingly :blush: :+1: